src/Game/Games/Hangman/Controller/ClientHangmanController.java

```diff
@@ -0,0 +1,120 @@
+package Game.Games.Hangman.Controller;
+
+import Contest.Model.ContestDataPersistor;
+import Game.Games.DataObject.HangmanDataObject;
+import Game.Games.DataObject.PlayerStatsDataObject;
+import Game.Games.Hangman.HangmanStatsEnum;
+import Game.Games.Hangman.Model.HangmanModel;
+import Game.Games.Hangman.View.HangmanView;
+import Online.Socket.Message.MessageDataObject;
+import Online.Socket.Message.MessageType;
+import Online.Socket.SocketCommunicatorService;
+import Player.Player;
+import Scene.Model.ActionEnum;
+
+import java.util.Map;
+import java.util.Observable;
+import java.util.Observer;
+
+public class ClientHangmanController extends HangmanController {
+
+    private final SocketCommunicatorService socketCommunicatorService;
+
+    public ClientHangmanController(HangmanModel model, HangmanView view, boolean isTraining, SocketCommunicatorService socketCommunicatorService) {
+        super(model, view, isTraining);
+
+        this.socketCommunicatorService = socketCommunicatorService;
+        this.socketCommunicatorService.addReceptionObserver(new SocketReceptionObserver());
+    }
+
+    private void updateGame(HangmanDataObject hangmanDataObject) {
+        ((HangmanView) this.view).setNumGuesses(hangmanDataObject.getResult());
+        ((HangmanView) this.view).setCurrentWord(hangmanDataObject.getWord());
+        ((HangmanView) this.view).setNumGuesses(hangmanDataObject.getNbGuess());
+        this.updateHangmanImage(hangmanDataObject.getGuessLeft());
+    }
+
+    private void updateHangmanImage(int guessLeft) {
+
+        if (guessLeft > 7) {
+            ((HangmanView) this.view).setImageIcon("/hw3/hang7.gif");
+        } else {
+            switch (guessLeft) {
+
+                case 7:
+                    ((HangmanView) this.view).setImageIcon("/data/Images/hang7.gif");
```

Comment:
instead of using a switch, you can do:
`((HangmanView) this.view).setImageIcon("/data/Images/hang" + guessLeft + ".gif");`

Reply:
Need to check